commons/src/main/java/org/infinispan/commons/marshall/jboss/ExtendedRiverMarshaller.java

```diff
@@ -35,4 +36,12 @@ void setCloseListener(RiverCloseListener listener) {
       this.listener = listener;
    }
 
+   int instanceCacheResizeCount() {
+      Object instanceCache = ReflectionUtil
+         .getValue(this, "instanceCache");
+
+      return (int) ReflectionUtil
+         .getValue(instanceCache, "resizeCount");
```

Comment:
I know `resizeCount` is the actual capacity, but I'd rather use `keys.length` (and call the method `instanceCacheCapacity`) instead because we care about memory usage here.